### PR TITLE
Make password retrieval robust/resilient

### PR DESCRIPTION
**Pull request by bheuvel**
If a job_result should be present, the code will now actually wait until it is actually present.
For readability of the main code, moved the code to a separate procedure.

### CODE REVIEW

File: lib/vagrant-cloudstack/action/run_instance.rb [hunk 1]
```diff
@@ -251,6 +241,34 @@ def call(env)
...
         end
 
+        def store_password(env, domain_config, job_id)
+          server_job_result = env[:cloudstack_compute].query_async_job_result({:jobid => job_id})
+          if server_job_result.nil?
+            env[:ui].warn(' -- Failed to retrieve job_result for retrieving the password')
+            return
+          end
+
+          while true
```
Comment (miguelaferreira) [inline lib/vagrant-cloudstack/action/run_instance.rb:251]: This kind of poling logic could be better implemented using the `Util::Timer` class. Please have a look at line 160 from`run_instance.rb`. Can you use the same approach here?

**Comment by miguelaferreira** [comment]
Functionally looks good, but the polling mechanism used when retrieving the password could be better implemented with the `Util::Timer` class.

Edit: retries are not implemented with `Unit::Timer`, they are created with a retryable and that triggers on errors, therefore it is not appropriate

File: lib/vagrant-cloudstack/action/run_instance.rb (hunk 1 above)
Comment (bheuvel) [inline lib/vagrant-cloudstack/action/run_instance.rb:251]: Check, better to timeout then an endless loop. Copied this from `create_port_forwarding_rule` and `create_firewall_rule` which can be improved as well :wink: Do we have a general cloud communication timeout value somewhere or use a specific timeout/parameter for each different operation?

Comment (miguelaferreira) [inline lib/vagrant-cloudstack/action/run_instance.rb:251]: The default VM timeout in vagrant is 300 seconds, but that is for the full deployment period.
I think that is too much for a single API call, maybe 10 seconds? 
Not really sure what make sense. I'll be happy with whatever you find reasonable.

**Pull request merged**